functions/v2/helloworld/hello-gcs/src/test/java/functions/HelloGcsTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package functions;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.common.testing.TestLogHandler;
+import com.google.gson.Gson;
+import functions.eventpojos.GcsEvent;
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.builder.CloudEventBuilder;
+import java.net.URI;
+import java.util.Date;
+import java.util.logging.Logger;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+/** Unit tests for main.java.com.example.functions.helloworld.HelloGcs. */
+public class HelloGcsTest {
+  private static final TestLogHandler LOG_HANDLER = new TestLogHandler();
+  private static final Logger logger = Logger.getLogger(HelloGcs.class.getName());
+
+  @Before
+  public void beforeTest() throws Exception {
+    logger.addHandler(LOG_HANDLER);
+  }
+
+  @After
+  public void afterTest() {
+    LOG_HANDLER.clear();
+  }
+
+  @Test
+  public void helloGcs_shouldPrintFileName() throws JsonProcessingException {
+    // Create event data
+    GcsEvent gcsEvent = new GcsEvent();
```

Comment:
Nit: I wonder if the CloudEvent type libraries (if we used them) would support the Builder pattern here.

Reply:
I don't believe the library is has that functionality either